frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -185,6 +214,82 @@ const MenuListItem = forwardRef<HTMLLIElement, MenuListItemProps>(
 function MainMenu(props: Props): ReactElement {
   const isServerDisconnected = !props.isServerConnected
 
+  const onClickDeployApp = useCallback((): void => {
+    const {
+      deployParams,
+      showDeployError,
+      closeDialog,
+      isDeployErrorModalOpen,
+    } = props
+
+    if (
+      (!deployParams ||
+        !deployParams.repository ||
+        !deployParams.branch ||
+        !deployParams.module) &&
+      !deployParams?.isHeadDetached
+    ) {
+      const dialog = NoRepositoryDetected()
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    if (deployParams?.isHeadDetached) {
```

Comment:
Curious if typescript is requiring the `?` cause you do check it does not exist in 225.

Think it might be nice since this is consistent to just do:
`const { isHeadDetached, module, untrackedFiles, uncommittedFiles, isAhead } = deployParams`. Saves a lot of `deployParams` checks.

There's probably a more clever way to make this easy, but I figure it is probably fine to just be explicit.

Reply:
done